### PR TITLE
feat: new accounts page

### LINKED ISSUE #2660: New account page wire up
### Overview

Hook up some new selectors to the new style account page

### References and additional details

Look for `AccountsV2`

### Acceptance Criteria

The new style page (that currently appears with the flag on in a non-functional way) behaves as expected.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

1 day

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

replaces the old account page with the new hierarchical colorful style supporting multiple accounts

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #2660

## Risk

* balances don't aggregate correctly across chains, accounts, or account numbers

## Testing

* this should appear with the multi account flag off (it's replacing the old one)
* the `Add Account` button should only appear with the flag on (and doesn't do anything currently)
* clicking the `Account #0` row for UTXO chains will not work - we do not have the concept of a route for "show me all the accounts for account number 0" yet
* clicking the `Account #0` row for account-based chains will work - and will take you to that individual account
* clicking on a token within an account-based chain (e.g. FOX) will take you to that account asset page
* clicking on a `Legacy/Segwit/Segwit Native` account within the dropdown will take you to that specific account

### Engineering

* see common testing steps

### Operations

* see common testing steps

## Screenshots (if applicable)

image

### CODE REVIEW

File: src/pages/Accounts/components/AccountNumberRow.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  accountIds: AccountId[]
+  chainId: ChainId
+} & ButtonProps
+
+type UtxoAccountEntriesProps = {
+  accountIds: AccountId[]
+  chainId: ChainId
+}
+
+const UtxoAccountEntries: React.FC<UtxoAccountEntriesProps> = ({ accountIds, chainId }) => {
```
Comment (0xdef1cafe) [inline src/pages/Accounts/components/AccountNumberRow.tsx:47]: needs a comment explaining why we're mapping `accountId`s here vs `assetIds`

File: src/state/slices/portfolioSlice/selectors.ts [hunk 2]
```diff
@@ -638,24 +675,22 @@ export const selectPortfolioAssetBalancesSortedFiat = createSelector(
 export const selectPortfolioAssetAccountBalancesSortedFiat = createSelector(
   selectPortfolioFiatAccountBalances,
   selectBalanceThreshold,
-  (
-    portfolioFiatAccountBalances,
-    balanceThreshold,
-  ): { [k: AccountSpecifier]: { [k: AssetId]: string } } => {
```
Comment (0xdef1cafe) [inline src/state/slices/portfolioSlice/selectors.ts:644]: i just aliased and inlined this type while i was here as i needed it elsewhere - this selector logic hasn't changed

File: src/state/slices/portfolioSlice/selectors.ts [hunk 3]
```diff
@@ -745,6 +799,12 @@ export const selectPortfolioAccountIdsSortedFiat = createDeepEqualOutputSelector
   },
 )
 
+export const selectPortfolioChainIds = createDeepEqualOutputSelector(
```
Comment (0xdef1cafe) [inline src/state/slices/portfolioSlice/selectors.ts:802]:
```suggestion
export const selectPortfolioChainIdsSortedFiat = createDeepEqualOutputSelector(
```

**Comment by 0xdef1cafe** [comment]
@shapeshift/operations if you could give this a once over it'd be great. this already has sign off from @DiggyDiggy2

**Comment by tshifty** [comment]
Ops is calling this PR g2g

File: src/pages/Accounts/components/AccountEntryRow.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  const history = useHistory()
+  const translate = useTranslate()
+  const filter = useMemo(() => ({ assetId, accountId }), [accountId, assetId])
+  const accountNumber = useAppSelector(s => selectAccountNumberByAccountId(s, filter))
+  const asset = useAppSelector(s => selectAssetById(s, assetId))
+  const cryptoBalance = useAppSelector(s => selectPortfolioCryptoHumanBalanceByFilter(s, filter))
+  const fiatBalance = useAppSelector(s => selectPortfolioFiatBalanceByFilter(s, filter))
+  const { icon, name, symbol } = asset
+
+  const isUtxoAccount = useMemo(() => isUtxoAccountId(accountId), [accountId])
```
Comment (gomesalexandre) [inline src/pages/Accounts/components/AccountEntryRow.tsx:33]: Nice 🍛 ser

File: src/state/slices/portfolioSlice/selectors.ts [hunk 5]
```diff
@@ -710,6 +747,25 @@ export const selectPortfolioTotalFiatBalanceByAccount = createSelector(
...
 )
 
+export const selectPortfolioAccountBalanceByAccountNumberAndChainId = createSelector(
+  selectPortfolioTotalFiatBalanceByAccount,
+  selectPortfolioAccountMetadata,
+  selectAccountNumberParamFromFilter,
+  selectChainIdParamFromFilter,
+  (accountBalances, accountMetadata, accountNumberString, chainId): string => {
+    const accountNumber = parseInt(accountNumberString.toString())
+    if (isNaN(accountNumber))
```
Comment (gomesalexandre) [inline src/state/slices/portfolioSlice/selectors.ts:757]: Normally would comment against use of NaN, but given `parseInt` was used to parse `accountNumberString`, this is fine 🤌

**Review by gomesalexandre — COMMENTED** [review]
Tested against prod, conceptually fine with this one minus a littlish regression, see below

- UTXOs balance is matching the total balance from all account types on prod
- Ethereum/Avalanche balance is matching the total balance from native asset + tokens
  - Individual native assets / tokens balances are matching the ones on prod
- Account types / assets are sorted by balance


Spotted one regression vs. develop/prod:

- Cosmos and Osmosis balances are slightly off, both on fiat and crypto value

image
image

File: src/state/slices/portfolioSlice/selectors.ts (hunk 5 above)
Comment (0xdef1cafe) [reply src/state/slices/portfolioSlice/selectors.ts:757]: as discussed lodash `isNumber` is a better pattern for this valid zero value 🤦

File: src/pages/Accounts/components/AccountNumberRow.tsx [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+    [accountAssetBalancesSortedFiat, accountId],
+  )
+  const entries = useMemo(
+    () =>
+      assetIds.map(assetId => (
+        <AccountEntryRow key={assetId} accountId={accountId} assetId={assetId} />
+      )),
+    [accountId, assetIds],
+  )
+  return <>{entries}</>
```
Comment (0xdef1cafe) [inline src/pages/Accounts/components/AccountNumberRow.tsx:75]: move inside memo

File: src/pages/Accounts/components/AccountNumberRow.tsx [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+const UtxoAccountEntries: React.FC<UtxoAccountEntriesProps> = ({ accountIds, chainId }) => {
+  const { assetId } = useAppSelector(s => selectFeeAssetByChainId(s, chainId))
+  const entries = useMemo(
+    () =>
+      accountIds.map(accountId => (
+        <AccountEntryRow key={`${assetId}-${accountId}`} accountId={accountId} assetId={assetId} />
+      )),
+    [accountIds, assetId],
+  )
+  return <>{entries}</>
```
[the remark this answers is not in the document]
Comment (0xdef1cafe) [reply src/pages/Accounts/components/AccountNumberRow.tsx:56]: move inside memo

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
LGTM overall, tested according to the previous testing criteria and this is still doing what it says on the box (minus one sorting bug, see below), including the inclusion of staking and delegation balances into the accounts values for Cosmos SDK accounts.

- Added one question for product and some terminology nitpicks.
- Also found one issue WRT UTXOs script type sorting, which are always sorted by zpub-ypub-xpub or equivalent for LTC, instead of being sorted by actual account balances:

image
image

File: src/pages/Accounts/components/ChainRow.tsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+  const accountRows = useMemo(() => {
+    return Object.entries(accountIdsByAccountNumber).map(([accountNumber, accountIds]) => (
+      <AccountNumberRow
+        key={accountNumber}
+        accountNumber={Number(accountNumber)}
+        accountIds={accountIds}
+        chainId={chainId}
+        onClick={
+          // accountIds is strictly length 1 per accountNumber for account-based chains
```
[the remark this answers is not in the document]
Comment (0xdef1cafe) [reply src/pages/Accounts/components/ChainRow.tsx:44]: agree, improvement (but not blocker) for @reallybeard when he's back 🖍️

File: src/pages/Accounts/components/ChainRow.tsx [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+type ChainRowProps = {
+  chainId: ChainId
+}
+
+export const ChainRow: React.FC<ChainRowProps> = ({ chainId }) => {
+  const { isOpen, onToggle } = useDisclosure()
+  const history = useHistory()
+  const asset = useAppSelector(s => selectFeeAssetByChainId(s, chainId))
+  const filter = useMemo(() => ({ chainId }), [chainId])
+  const chainFiatBalance = useAppSelector(s => selectPortfolioFiatBalanceByChainId(s, filter))
```
[the remark this answers is not in the document]
Comment (0xdef1cafe) [reply src/pages/Accounts/components/ChainRow.tsx:29]: i think the selector name gives sufficient context as to not make the local variable too verbose, but i could go either way - up to you ser

[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/pages/Accounts/components/ChainRow.tsx:29]: Let's keep this one, agree it's shorter and easy to understand alongside the selector name - one can go to the definition of the selector if in doubt what this is

**Comment by 0xdef1cafe** [comment]
@gomesalexandre https://github.com/shapeshift/web/pull/2667/commits/ac508492b6659a5995219266d465b0a836640e3d 

this just uses the new nested sorted object to grab the sorted account ids, fixes the sorting within the account number aggregation for BTC and LTC

**Review by gomesalexandre — APPROVED** [review]
LGTM, tested again, all review comments have been tackled and the script type sorting bug has been fixed

🍝

**Pull request merged**